src/librustc_const_eval/eval.rs

```diff
@@ -482,6 +469,7 @@ impl ConstEvalErr {
             },
             BadType(ref i) => format!("value of wrong type: {:?}", i).into_cow(),
             ErroneousReferencedConstant(_) => "could not evaluate referenced constant".into_cow(),
+            BadCharValue => "invalid numeric value for char".into_cow(),
```

Comment:
Could you add tests for this warnings? Probably `some_i8 as char` and `u32_thats_not_a_valid_char as char`